hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -136,7 +136,15 @@ def get_type_hints(thing):
         NameError for unresolvable forward references, just return an empty dict.
         """
         try:
-            hints = typing.get_type_hints(thing)
+            if inspect.isclass(thing) and hasattr(thing, "__signature__"):
+                spec = inspect.getfullargspec(thing)
+                hints = {
+                    k: v
+                    for k, v in spec.annotations.items()
+                    if k in (spec.args + spec.kwonlyargs) and isinstance(v, type)
```

Comment:
Why don't we use `varargs, varkw` here?

Reply:
I was just following the pattern as used in the above version. Are you suggesting something like this:
```python
                all_args = (
                    spec.args
                    + (spec.varargs or [])
                    + (spec.varkw or [])
                    + spec.kwonlyargs
                )
                hints = {
                    k: v
                    for k, v in spec.annotations.items()
                    if k in all_args and isinstance(v, type)
                }
```

Reply:
We skip varargs and kwargs, because we always generate zero of them as our inferred strategy - similar to preferring to use the default value (if there is one) rather than type hints.

Changing this would involve a LOT of other work messing with signatures, and is well out of scope for the foreseeable future.